### PR TITLE
application/x-protobuf MIME type as compressible

### PR DESCRIPTION
**Pull request by reyemtm**
Used for vector tiles in web mapping applications.

### CODE REVIEW

File: src/custom-types.json [hunk 1]
```diff
@@ -347,7 +347,11 @@
   "application/x-pkcs12": {
     "compressible": false
   },
-  "application/x-rar-compressed": {
+  "application/x-protobuf": {
+    "compressible": true,
+    "sources": ["https://developers.google.com/api-client-library/java/google-http-java-client/reference/1.20.0/com/google/api/client/protobuf/ProtocolBuffers"]
+  },
+    "application/x-rar-compressed": {
```
Comment (dougwilson) [inline src/custom-types.json]: Indent error here

File: src/custom-types.json [hunk 2]
```diff
@@ -347,7 +347,11 @@
   "application/x-pkcs12": {
     "compressible": false
   },
-  "application/x-rar-compressed": {
+  "application/x-protobuf": {
+    "compressible": true,
+    "sources": ["https://developers.google.com/api-client-library/java/google-http-java-client/reference/1.20.0/com/google/api/client/protobuf/ProtocolBuffers"]
```
Comment (dougwilson) [inline src/custom-types.json]: Is there a source you can add about the compressibility of the type?

**Comment by reyemtm** [comment]
Hmm I did not find anything on that initially. I'll have to do some digging.

File: src/custom-types.json (hunk 1 above)
Comment (reyemtm) [reply src/custom-types.json]: not sure what you mean here - I changed the url to be in a new line if that's what you mean

File: src/custom-types.json (hunk 2 above)
Comment (reyemtm) [reply src/custom-types.json]: There is lots of discussion on the [mapbox vector tile spec github issues page](https://github.com/mapbox/vector-tile-spec/issues/27) on compression, but a more authoritative source which clearly states that vector tiles (which are protocol buffers) can be compressed or uncompressed, is the GDAL documentation on [Mapbox Vector Tiles](https://gdal.org/drivers/vector/mvt.html).

File: src/custom-types.json (hunk 1 above)
Comment (dougwilson) [reply src/custom-types.json]: It seems like there are four spaces now on this line when there used to be only 2 spaces.

**Comment by ctaylor4874** [comment]
Is there anything I can do to help with this PR? I'm also in need of adding `application/x-protobuf` to the compressible types.

**Comment by dougwilson** [comment]
Hi @ctaylor4874 thanks for the ping here. So the PR has an open formatting issue that was never resolved, and the including link results in a 404. But upon further research, it seems that `application/x-protobuf` is a binary container format. So by itself it is not compressible due to the nature of the way in which it encodes the values. That said, of course perhaps the contents in general may be compressible if there are large, compressible strings inside the protobuf contents, just like how a ZIP file can be compressed if the files inside it are compressible but not actually compressed.

Due to this, the actual MIME type `application/x-protobuf` cannot be marked as either yes definately compressible or no definately not compressible in this module, as the MIME type does not provide enough context for the likely compressibility.

**Pull request closed without merging**